From 7fb2642827387a9d5b99a873477eecc5b3fb7ed8 Mon Sep 17 00:00:00 2001
From: Holger Hans Peter Freyther <zecke@selfish.org>
Date: Mon, 19 Jan 2009 00:37:23 +0100
Subject: [PATCH] Initialize threading support...

---
 WebKitTools/GtkLauncher/main.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/WebKitTools/GtkLauncher/main.c b/WebKitTools/GtkLauncher/main.c
index 1067445..fc59a9b 100644
--- a/WebKitTools/GtkLauncher/main.c
+++ b/WebKitTools/GtkLauncher/main.c
@@ -341,6 +341,7 @@ create_window ()
 int
 main (int argc, char* argv[])
 {
+    g_thread_init (NULL);
     gtk_init (&argc, &argv);
     if (!g_thread_supported ())
         g_thread_init (NULL);
-- 
1.5.6.3

