From 496b28397b8697f96145e89f2a027f31375025d9 Mon Sep 17 00:00:00 2001
From: Holger Hans Peter Freyther <zecke@selfish.org>
Date: Wed, 18 Mar 2009 17:55:35 +0800
Subject: [PATCH] Start changing the stylesheet to hide things...

---
 public/stylesheets/application.css |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/public/stylesheets/application.css b/public/stylesheets/application.css
index dae40eb..891bd67 100644
--- a/public/stylesheets/application.css
+++ b/public/stylesheets/application.css
@@ -198,3 +198,7 @@ table.admin {
 
 .floatright {
   float: right; }
+
+span.flagicon {
+  display: none;
+}
-- 
1.5.6.3

