From 574f016c50ca01ccbe038d7d1a2d77b1d2dce5e5 Mon Sep 17 00:00:00 2001
From: Holger Freyther <ich@imp050244.vpn.mi.fu-berlin.de>
Date: Mon, 19 Jan 2009 12:11:54 +0100
Subject: [PATCH] Change column count to one for our device

---
 skins/MonoBook.php |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/skins/MonoBook.php b/skins/MonoBook.php
index bff7d39..fc5d1b2 100644
--- a/skins/MonoBook.php
+++ b/skins/MonoBook.php
@@ -145,6 +145,15 @@ for (i  = 0; i < nodes.length; ++i) {
         nodes[i].style.setProperty("display", "none")
     }
 }
+
+// Twist things around....
+var nodes = document.querySelectorAll("div.references-small")
+for (i = 0; i < nodes.length; ++i) {
+    var column = nodes[i].style.getPropertyValue("-webkit-column-count")
+    if (column != null) {
+        nodes[i].style.setProperty("-webkit-column-count", "1")
+    }
+}
 </script>
 </body></html>
 <?php
-- 
1.6.0.2

