From a365d634eef73ca869632773216c0766857642b8 Mon Sep 17 00:00:00 2001
From: Holger Hans Peter Freyther <zecke@selfish.org>
Date: Sat, 20 Dec 2008 22:22:01 +0100
Subject: [PATCH] Use len attribute of GByteArray instead of strlen (which stops at the first null byte)

---
 WebKitTools/GtkLauncher/main.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/WebKitTools/GtkLauncher/main.c b/WebKitTools/GtkLauncher/main.c
index 69f95b9..0162462 100644
--- a/WebKitTools/GtkLauncher/main.c
+++ b/WebKitTools/GtkLauncher/main.c
@@ -57,7 +57,7 @@ static guint status_context_id;
     g_free(c_content_##var);                                      \
     } while(0);
 
-#if 0
+#if 1
 #define WRITE_BITMAP(path, glyph, var, width, height) \
     do {                                                          \
     GString* path_##var = g_string_new(path);                     \
@@ -75,10 +75,11 @@ static guint status_context_id;
                             (guint)glyph->width);                 \
     }                                                             \
     gchar* c_path_##var = g_string_free(path_##var, FALSE);       \
+    int length = content_##var->len;                              \
     gchar* c_content_##var = g_byte_array_free(content_##var, FALSE); \
     GError* error = NULL;                                         \
     g_file_set_contents(c_path_##var, c_content_##var,            \
-                        strlen(c_content_##var), &error);         \
+                        length, &error);                          \
     if (error) {                                                  \
         g_warning("failed to write '%s'", c_path_##var);          \
         g_error_free(error);                                      \
@@ -148,6 +149,7 @@ static void create_font_dirs(GList *glyphs)
  */
 static void create_text_file(GList *glyphs)
 {
+
 }
 
 static void create_links(GList *links)
-- 
1.5.6.3

