From 58104e974e53833a88f7fe5587de9bbc5f63c8dc Mon Sep 17 00:00:00 2001
From: Holger Hans Peter Freyther <zecke@selfish.org>
Date: Thu, 5 Mar 2009 19:27:18 +0800
Subject: [PATCH] Use the pitch...

---
 WebKitTools/GtkLauncher/main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/WebKitTools/GtkLauncher/main.c b/WebKitTools/GtkLauncher/main.c
index d8627e5..d2b8c42 100644
--- a/WebKitTools/GtkLauncher/main.c
+++ b/WebKitTools/GtkLauncher/main.c
@@ -74,7 +74,7 @@ static gint dumped_glyphs = 0;
     int row = 0;                                                  \
     for (row = 0; row < glyph->height; ++row) {                   \
         g_byte_array_append(content_##var,                        \
-                            glyph->bitmap + (row*glyph->width),   \
+                            glyph->bitmap + (row*glyph->glyph_pitch),\
                             (guint)glyph->width);                 \
     }                                                             \
     gchar* c_path_##var = g_string_free(path_##var, FALSE);       \
-- 
1.5.6.3

