From c07c463b59741186d4c4b178d88f9da36337bc4a Mon Sep 17 00:00:00 2001
From: Holger Freyther <ich@imp050244.vpn.mi.fu-berlin.de>
Date: Mon, 19 Jan 2009 13:43:11 +0100
Subject: [PATCH 21/22] Add Hyphenator script to hyphenate the content..

---
 skins/MonoBook.php |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/skins/MonoBook.php b/skins/MonoBook.php
index e42bb7c..0d667e8 100644
--- a/skins/MonoBook.php
+++ b/skins/MonoBook.php
@@ -155,6 +155,11 @@ for (i = 0; i < nodes.length; ++i) {
     }
 }
 </script>
+<script src="http://hyphenator.googlecode.com/svn/trunk/Hyphenator.js" type="text/javascript"></script>
+<script type="text/javascript">
+    Hyphenator.setMinWordLength(4);
+    Hyphenator.run();
+</script>
 </body></html>
 <?php
 	wfRestoreWarnings();
-- 
1.6.1

